src/segment.go

```diff
@@ -112,15 +112,45 @@ func (segment *Segment) getValue(property Property, defaultValue string) string
 	return defaultValue
 }
 
-func (segment *Segment) shouldIgnoreFolder(cwd string) bool {
-	if value, ok := segment.Properties[IgnoreFolders]; ok {
-		list := parseStringArray(value)
-		for _, element := range list {
-			pattern := fmt.Sprintf("^%s$", element)
-			matched := matchString(pattern, cwd)
-			return matched
+func (segment *Segment) shouldIncludeFolder(cwd string) bool {
+	cwdIncluded := segment.cwdIncluded(cwd)
+	cwdExcluded := segment.cwdExcluded(cwd)
+	return (cwdIncluded && !cwdExcluded)
+}
+
+func (segment *Segment) cwdIncluded(cwd string) bool {
+	value, ok := segment.Properties[IncludeFolders]
+	if !ok {
+		// IncludeFolders isn't specified, everything is included
+		return true
+	}
+
+	list := parseStringArray(value)
+
+	if len(list) == 0 {
```

Comment:
This check isn't found anymore in exclude. Can be part of `cwdMatchesOneOf`

Reply:
You've suggested this twice now. Do you mean making `cwdMatchesOneOf` return two values? Or are you suggesting have `cwdMatchesOneOf` return true or false?  The reason this was never part of `cwdMatchesOneOf` is the behavior for an undefined set is different if we're talking about Include or Exclude.

Reply:
The way I see it `cwdMatchesOneOf ` works the same regardless of the context. Same goes for `cwdIncluded ` and `cwdExcluded `, they are the same apart from the array they check (if it wasn't for `IgnoreFolders ` this could be the same function. So the Len check had me confused, I would've expected it everywhere. Adding that to `cwdMatchesOneOf ` would return false in case `cwdIncluded ` meaning and empty array would not display the segment. I believe that's actually correct. No array means it returns true,  an empty array implies a mistake.

Reply:
> cwdMatchesOneOf works the same regardless of the context. 
Agreed

> Same goes for cwdIncluded and cwdExcluded, they are the same apart from the array they check 
Sorry, but you're mistaken here.  The difference lies in the code you're saying is not consistent: the implied value for an undefined property. 

cwdIncluded:  If the IncludeFolders property is not defined, everything is included (cwdIncluded returns true). 

cwdExcluded: If ExcludedFolders is not defined, nothing is excluded (cwdExcluded returns false).  I could make this more verbose in cwdExcluded, to explicitly return false, but it would not change the result (and would definitely not make these two functions be identical).

> No array means it returns true, an empty array implies a mistake.
I disagree here too. I think these two cases should be treated the same.  For cwdIncluded these two cases mean everything is included (return true).  For cwdExcluded these two cases means nothing is excluded (return false)

Reply:
Alright agreed, maybe it's more about how I would write this rather than the actual logic. And that's not something to argue over. Great work, this is very understandable and well argumented ;-)